### PR TITLE
Function to start, stop, restart thrift metastore server

### PR DESCRIPTION
**Pull request by akaranjkar-qu**

### CODE REVIEW

File: hive/thrift-metastore.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+source /usr/lib/hustler/bin/qubole-bash-lib.sh
+
+##
+# Start thrift metastore server
+#
+function start_thrift_metastore() {
+    is_master=$(nodeinfo is_master)
+    if [[ "$is_master" == "1" ]]; then
+        /usr/lib/hive1.2/bin/thrift-metastore server start
```
Comment (nitinik) [inline hive/thrift-metastore.sh]: We will have to take care of the hive version as well. Refer hadoop_node_init_ext.sh:start_hive_metastore_server()

File: hive/thrift-metastore.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    fi
+}
+
+##
+# Stop thrift metastore server
+#
+function stop_thrift_metastore() {
+    is_master=$(nodeinfo is_master)
```
Comment (nitinik) [inline hive/thrift-metastore.sh]: Refer https://github.com/qubole/bootstrap-functions/pull/4/files . Lets have common methods to check cluster_type , role, etc at one place.

File: hive/thrift-metastore.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    if [[ "$is_master" == "1" ]]; then
+        /usr/lib/hive1.2/bin/thrift-metastore server start
+    fi
+}
+
+##
+# Stop thrift metastore server
+#
+function stop_thrift_metastore() {
```
Comment (nitinik) [inline hive/thrift-metastore.sh:25]: Add a comment that this will only work for Hadoop2 clusters.

File: hive/thrift-metastore.sh (hunk 1 above)
Comment (akaranjkar-qu) [reply hive/thrift-metastore.sh]: Done

**Comment by hiyer** [comment]
@nitinik kindly approve the PR when it looks ok to you and I'll merge it.

File: hive/thrift-metastore.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    fi
+fi
+
+
+##
+# Start thrift metastore server
+#
+function start_thrift_metastore() {
+    if [[ "$is_master" == "1" ]]; then
+        ${thrift_metastore_bin_path}/thrift-metastore server start
```
Comment (shreya668) [inline hive/thrift-metastore.sh]: As discussed, lets start it through monit, rather than taking a dependency on bin path.

Comment (akaranjkar-qu) [reply hive/thrift-metastore.sh]: Done

File: hive/thrift-metastore.sh [hunk 5]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+source /usr/lib/hustler/bin/qubole-bash-lib.sh
+
+is_master=$(nodeinfo is_master)
```
Comment (himanshu-mishra) [inline hive/thrift-metastore.sh]: As https://github.com/qubole/bootstrap-functions/blob/master/common/utils.sh is now available, we should use that, and create new methods for thrift_metastore, hive_version

File: hive/thrift-metastore.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+source /usr/lib/qubole/bootstrap-functions/common/utils.sh
```
Comment (harshashah16) [inline hive/thrift-metastore.sh:3]: Can we remove populate_nodeinfo function or call it always for all NBL scripts?

**Review by harshashah16 — APPROVED** [review]
Approving as per Spandana's talk

File: hive/thrift-metastore.sh (hunk 6 above)
Comment (akaranjkar-qu) [reply hive/thrift-metastore.sh:3]: Added populate_nodeinfo in /usr/lib/qubole/bootstrap-functions/common/utils.sh in https://github.com/qubole/bootstrap-functions/pull/26

**Pull request merged**